src/square.tsx

```diff
@@ -0,0 +1,7 @@
+
+
+function Square({value=null, onSquareClick, won}) {
```

Comment:
playing with the basics of react I'm unable to add meaningful type defs here. 
not even the most basic type declaration, such as a boolean, works `function Square({value=null, onSquareClick, won:boolean }) {`
what am I missing @ivy-lli?

Reply:
react component properties is one object, e.g.: 
```jsx
function Square (props) {}
```

To define the type of the object you need to define the whole object, e.g.: 
```tsx
function Square (props: {value: string, onSquareClick: () => void, won: boolean}) {
  props.value
}
```

If you don't like to access values via `props` you can do what I think you intended to do above, descructure the object. Or better access the single values in the object directly:
```tsx
function Square ({value, onSquareClick, won}: {value: string, onSquareClick: () => void, won: boolean}) {
  value
}
```

At one point the line will be very big, this is normally the time when I create a separate type for the props, e.g.:
```tsx
type SquareProps = {value: string, onSquareClick: () => void, won: boolean};

function Square ({value, onSquareClick, won}: SquareProps) {
  value
}
```

What you also can do, e.g. if you only need the value in this component and want to drill the other props to another component in it, you can only descructure some of the properties and put all the rest into a new property. But then the properties of the "inner" component needs the have the properties as the same name:
```tsx
type SquareProps = SomeOtherComponentProps & {value: string};

function Square ({value, ...props}: SquareProps) {
  return <SomeOtherComponent {...props}>{value}</SomeOtherComponent>
}

type SomeOtherComponentProps = {onSquareClick: () => void, won: boolean};

function SomeOtherComponent (props: SomeOtherComponentProps) {}
```

Reply:
Thanks a lot @ivy-lli 
I've sort of documented almost all variants in code now and am going to digest it later on
https://github.com/ivy-rew/vite-tictactoe-tutorial/pull/1/commits/466207dc824d139865f040e682f3716fda7c3207